java/src/com/ibm/streamsx/topology/kafka/KafkaStreams.java

```diff
@@ -0,0 +1,284 @@
+     * @param kafkaConsumerConfig see the Kafka documentation.
+     *        Minimal configuration typically includes:
+     * <ul>
+     * <li><code>zookeeper.connect</code></li>
+     * <li><code>group.id</code></li>
+     * <li><code>zookeeper.session.timeout.ms</code></li>
+     * <li><code>zookeeper.sync.time.ms</code></li>
+     * <li><code>auto.commit.interval.ms</code></li>
+     * </ul>
+     * @param threadsPerTopic number of threads to allocate to processing each topic
+     * @param topics one or more Kafka topics to subscribe to
+     * @return TStream&lt;KafkaTuple>
+     *      The generated tuples do not have a {@code topic}.
+     * @throws IllegalArgumentException if topics is null or empty.
+     * @throws IllegalArgumentException if threadsPerTopic <= 0.
+     */
+    public static TStream<KafkaMessage> source(TopologyElement te, Properties kafkaConsumerConfig,
+            int threadsPerTopic,
+            String... topics) {
+        
+        if (topics==null || topics.length==0)
+            throw new IllegalArgumentException("topics");
+        if (threadsPerTopic <= 0)
+            throw new IllegalArgumentException("threadsPerTopic");
+
+        Map<String, Object> params = new HashMap<>();
+        params.put("topic", topics);
+        params.put("threadsPerTopic", threadsPerTopic);
+        params.put("propertiesFile", PROP_FILE);
+        if (kafkaConsumerConfig!=null && !kafkaConsumerConfig.isEmpty())
+            params.put("kafkaProperty", toKafkaProperty(kafkaConsumerConfig));
+        
+        addPropertiesFile(te);
+
+        boolean confirmedMultiTopicConsumerWorks = false; // seems broken
+        if (topics.length==1 || confirmedMultiTopicConsumerWorks) {
+            SPLStream rawKafka = SPL.invokeSource(
+                            te,
+                            "com.ibm.streamsx.messaging.kafka::KafkaConsumer",
+                            params,
+                            KafkaSchemas.KAFKA);
+
+            return toKafkaTupleStream(rawKafka);
+        }
+        else {
+            // fake it with N kafkaConsumers
+            List<TStream<KafkaMessage>> list = new ArrayList<TStream<KafkaMessage>>(topics.length);
+            for (String topic : topics) {
+                Map<String,Object> myParams = new HashMap<>(params);
+                myParams.put("topic", topic);
+
+                SPLStream rawKafka = SPL.invokeSource(
+                        te,
+                        "com.ibm.streamsx.messaging.kafka::KafkaConsumer",
+                        myParams,
+                        KafkaSchemas.KAFKA);
+                
+                list.add(toKafkaTupleStream(rawKafka));
+            }
+            TStream<KafkaMessage> kafka = list.remove(0);
+            
+            return kafka = kafka.union(new HashSet<>(list));
+        }
+    }
+    
+    /**
+     * Same as {@code source(te, topic, 1, kafkaConsumerConfig)}
```

Comment:
Have the summary describe what it does. In the body you can say it is equivalent to another method.